### ISSUE TITLE
Support for loading Pango into wxTextCtrl (wxGTK3)

### ISSUE THREAD

**Issue by Blake-Madden** [wxWidgets/wxWidgets#24912]
Akin to the recent `wxTextCtrl` RTF loading (and saving) support added to the MSW and OSX ports, would there be interest in a similar API for Pango in the GTK3 port?

`gtk_text_buffer_get_text` and `gtk_text_buffer_insert_markup` (added in GTK+ 3.16) provide the ability to get and set Pango content to a text control. Similar to the RTF functions, doing this provides a much more optimal way to create large, highly-formatted content first and then load it into the control.

If there is interest, I was thinking the API should be wxGTK3 specific. So instead of `SetPangoValue()` being in the base class, add `GTKSetPangoValue()` and `GTKGetPangoValue()` to the GTK implementation instead.

**Comment by vadz** [comment]
I wonder if there is a simple way to convert Pango markup to RTF? If there were, we could provide `SetMarkup()` under all platforms, converting to RTF if necessary.

**Comment by Blake-Madden** [comment]
Perhaps not simple, but it could be possible :) At the very least, the subset of Pango that `wxControl::SetLabelMarkup` supports would be really nice.

Should I go ahead and make a GTK-specific `GTKSetPangoValue()` PR and then explore a base `SetMarkup()` function that would wrap `GTKSetPangoValue()` or `SetRFTValue()`, depending on platform? I would probably need to hand roll a convertor for Pango-subset -> RTF, but hopefully that wouldn't be too bad.

**Comment by vadz** [comment]
Yes, sure, adding `GTKSetPangoMarkup()` (I prefer this to a more generic "value") should be simple enough and I don't see any reason _not_ to do it.

TIA!

**Comment by ChipZhang** [comment]
`gtk_text_buffer_insert_markup()` is introduced in GTK 3.16, so you should check for GTK version. Or there will be compile time or run time errors if the GTK version is old.

Here is my prososal:

```cpp
void wxTextCtrl::GTKSetPangoMarkup(const wxString& str)
{
    wxCHECK_RET(IsMultiLine(), "pango markup requires multiline control");

#if GTK_CHECK_VERSION(3,16,0)
	if (gtk_check_version(3,16,0) != nullptr) {
		wxLogError(wxT("pango markup requires GTK+ 3.16.0+"));
		return;
	}
    // multiple events may get fired while editing text, so block those
    {
        EventsSuppressor noevents(this);
        // clear current content
        GtkTextIter start, end;
        gtk_text_buffer_get_bounds(m_buffer, &start, &end);
        gtk_text_buffer_delete(m_buffer, &start, &end);

        gtk_text_buffer_insert_markup(m_buffer, &start, str.utf8_str(), -1);
    }
    SendTextUpdatedEvent(GetEditableWindow());
#else
	wxLogError(wxT("pango markup requires GTK+ 3.16.0+"));
#endif // GTK_CHECK_VERSION(3,16,0)
}
```

This is the diff:
```diff
> #if GTK_CHECK_VERSION(3,16,0)
> 	if (gtk_check_version(3,16,0) != nullptr) {
> 		wxLogError(wxT("pango markup requires GTK+ 3.16.0+"));
> 		return;
> 	}
2132a2138,2140
> #else
> 	wxLogError(wxT("pango markup requires GTK+ 3.16.0+"));
> #endif // GTK_CHECK_VERSION(3,16,0)
```

**Comment by ChipZhang** [comment]
@vadz @Blake-Madden
